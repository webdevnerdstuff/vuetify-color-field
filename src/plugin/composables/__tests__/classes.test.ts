import { describe, it, expect } from 'vitest';
import {
	usePipClasses,
	useTextFieldClasses,
	useHintClasses,
	useCardClasses
} from '../classes';


describe('Classes Composable', () => {

	describe('usePipClasses', () => {
		it('should return class object', () => {
			const classes = usePipClasses();

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useTextFieldClasses', () => {
		it('should return class object', () => {
			const classes = useTextFieldClasses({
				readonlyInput: false,
			});

			expect(classes).toMatchInlineSnapshot(`
				{
				  "v-color-field--text-field": true,
				  "v-color-field--text-field-": false,
				  "v-color-field--text-field-readonly": false,
				  "v-color-field--text-field-readonly-input": false,
				}
			`);
		});

		it('should return class object', () => {
			const classes = useTextFieldClasses({
				name: 'foobar',
				readonly: true,
				readonlyInput: true,
			});

			expect(classes).toMatchInlineSnapshot(`
				{
				  "v-color-field--text-field": true,
				  "v-color-field--text-field-foobar": true,
				  "v-color-field--text-field-readonly": true,
				  "v-color-field--text-field-readonly-input": false,
				}
			`);
		});
	});

	describe('useHintClasses', () => {
		it('should return class object', () => {
			const classes = useHintClasses({

			});

			expect(classes).toMatchSnapshot();
		});
	});

	describe('useCardClasses', () => {
		it('should return class object', () => {
			const classes = useCardClasses({
				fullWidth: false,
			});

			expect(classes).toMatchSnapshot();
		});
	});

	it('should return class object', () => {
		const classes = useCardClasses({
			fullWidth: true,
		});

		expect(classes).toMatchSnapshot();
	});
});
