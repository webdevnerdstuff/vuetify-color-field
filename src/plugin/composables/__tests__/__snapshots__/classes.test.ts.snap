// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Classes Composable > should return class object 1`] = `
{
  "v-color-field--card": true,
  "v-color-field--card-full-width": true,
}
`;

exports[`Classes Composable > useCardClasses > should return class object 1`] = `
{
  "v-color-field--card": true,
  "v-color-field--card-full-width": false,
}
`;

exports[`Classes Composable > useHintClasses > should return class object 1`] = `
{
  "v-color-field--text-field": true,
}
`;

exports[`Classes Composable > usePipClasses > should return class object 1`] = `
{
  "v-color-field--pip": true,
}
`;
